src/LoanManager.ts

```diff
@@ -0,0 +1,171 @@
+import BN from "bn.js";
+import { LoanManager as LoanManagerInstance } from "../generated/index";
+import { LoanManager_ABI_ec709c3341045caa3a75374b8cfc7286 as LegacyLoanManagerInstanceWithChunkSize } from "../generated/types/LoanManager_ABI_ec709c3341045caa3a75374b8cfc7286";
+// import { TransactionObject } from "../generated/types/types";
+import { AbstractContract } from "./AbstractContract";
+import { AugmintToken } from "./AugmintToken";
+import { BN_PPM_DIV, CHUNK_SIZE, LEGACY_CONTRACTS_CHUNK_SIZE, MIN_LOAN_AMOUNT_ADJUSTMENT, PPM_DIV } from "./constants";
+import { EthereumConnection } from "./EthereumConnection";
+import { Rates } from "./Rates";
+// import { Transaction } from "./Transaction";
+
+export interface ILoanManagerOptions {
+    token: AugmintToken;
+    rates: Rates;
+    ethereumConnection: EthereumConnection;
+}
+
+export interface ILoanProduct {
```

Comment:
Should be a class, so you can implement various helpful calculation methods here.